File: siteMusifan/src/main/java/siteMusifan/entity/Concert.java
package siteMusifan.entity;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "concert")
@SequenceGenerator(name = "seqConcert", sequenceName = "seq_concert", allocationSize = 1, initialValue = 100)
public class Concert {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqConcert")
	@Column(name = "concert_id")
	@JsonView({JsonViews.Common.class,})
	private Long id; 
	@Column(name = "concert_nom")
	@JsonView({JsonViews.Common.class,})
	private String nom;
	@Column(name = "concert_date")
	@JsonView({JsonViews.Common.class,})
	private LocalDate date = LocalDate.now();
	@ManyToOne
	@JoinColumn(name = "concert_lieu_id", foreignKey = @ForeignKey(name = "concert_lieu_id_fk"))
	@JsonView({JsonViews.ConcertAvecLieu.class,})
	private Lieu lieu;
	@Column(name = "concert_nbPlace")
	@JsonView({JsonViews.Common.class,})
	private int nbPlace;
	@Column(name = "concert_prix")
	@JsonView({JsonViews.Common.class,})
	private int prix;
	
	@Version
	@Column(name="concert_version")
	private int version;
	
	@OneToMany(mappedBy = "id.concert")
	private Set<LigneCommande> lignesCommandes;

	@OneToMany(mappedBy = "id.concert")
	private Set<LigneConcert> ligneConcerts = new HashSet<LigneConcert>();

	public Concert() {

	}
	
	public Concert(String nom, LocalDate date, Lieu lieu, int nbPlace, int prix) {
		this.nom = nom;
		this.date = date;
		this.lieu = lieu;
		this.nbPlace = nbPlace;
		this.prix = prix;
	}

	public Concert(String nom, LocalDate date, int nbPlace, int prix) {
		this.nom = nom;
		this.date = date;
		this.nbPlace = nbPlace;
		this.prix = prix;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}

	public Lieu getLieu() {
		return lieu;
	}

	public void setLieu(Lieu lieu) {
		this.lieu = lieu;
	}

	public int getNbPlace() {
		return nbPlace;
	}

	public void setNbPlace(int nbPlace) {
		this.nbPlace = nbPlace;
	}

	public int getPrix() {
		return prix;
	}

	public void setPrix(int prix) {
		this.prix = prix;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	public Set<LigneCommande> getLignesCommandes() {
		return lignesCommandes;
	}

	public void setLignesCommandes(Set<LigneCommande> lignesCommandes) {
		this.lignesCommandes = lignesCommandes;
	}

	public Set<LigneConcert> getLigneConcerts() {
		return ligneConcerts;
	}

	public void setLigneConcerts(Set<LigneConcert> ligneConcerts) {
		this.ligneConcerts = ligneConcerts;
	}
	
	//Ajouter un concert a la liste de concert d'un artiste
	public void addArtiste(Artiste artiste) {
		ligneConcerts.add(new LigneConcert(new LigneConcertPK(this, artiste)));
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Concert other = (Concert) obj;
		return Objects.equals(id, other.id);
	}

}

File: musifanAngular/src/app/models/concert.ts
import { Artiste } from './artiste';
import { Lieu } from './lieu';

export class Concert {
  public constructor(
    private _id?: number | undefined,
    private _nom?: string | undefined,
    private _date?: Date | undefined,
    private _nbPlace?: number | undefined,
    private _prix?: number | undefined,
    private _lieu?: Lieu | undefined,
    private _artistes?: Artiste[] | undefined
  ) {}

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Getter nom
   * @return {string }
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Setter nom
   * @param {string } value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Getter date
   * @return {Date }
   */
  public get date(): Date | undefined {
    return this._date;
  }

  /**
   * Setter date
   * @param {Date } value
   */
  public set date(value: Date | undefined) {
    this._date = value;
  }

  /**
   * Getter nbPlace
   * @return {number }
   */
  public get nbPlace(): number | undefined {
    return this._nbPlace;
  }

  /**
   * Setter nbPlace
   * @param {number } value
   */
  public set nbPlace(value: number | undefined) {
    this._nbPlace = value;
  }

  /**
   * Getter prix
   * @return {number }
   */
  public get prix(): number | undefined {
    return this._prix;
  }

  /**
   * Setter prix
   * @param {number } value
   */
  public set prix(value: number | undefined) {
    this._prix = value;
  }

  /**
   * Getter lieu
   * @return {Lieu }
   */
  public get lieu(): Lieu | undefined {
    return this._lieu;
  }

  /**
   * Setter lieu
   * @param {Lieu } value
   */
  public set lieu(value: Lieu | undefined) {
    this._lieu = value;
  }

  /**
   * Getter artistes
   * @return {Artiste[] }
   */
  public get artistes(): Artiste[] | undefined {
    return this._artistes;
  }

  /**
   * Setter artistes
   * @param {Artiste[] } value
   */
  public set artistes(value: Artiste[] | undefined) {
    this._artistes = value;
  }
}
